### PR TITLE
Execution Manager refactor

### PR DESCRIPTION
**Pull request by rekksson**
Execution Manager splited into Autosar EM application and EM server adapter.

### CODE REVIEW

File: applications/ExecutionManager/src/execution_manager.cpp [hunk 1]
```diff
@@ -27,47 +26,28 @@ namespace {
...
 
+const MachineState ExecutionManager::defaultState {"Startup"};
+
 ExecutionManager::ExecutionManager(std::unique_ptr<IManifestReader> reader)
   : m_activeApplications{}
-  , m_allowedApplicationForState{reader->getStatesSupportedByApplication()}
-  , m_currentState{}
-  , m_machineManifestStates{reader->getMachineStates()}
+  , m_allowedApplicationForState{
+      std::move(reader->getStatesSupportedByApplication())}
```
Comment (RomanLevkovych) [inline applications/ExecutionManager/src/execution_manager.cpp]: compiler would optimize this piece, so we don't need explicit `std::move` here

File: applications/ExecutionManager/src/execution_manager.cpp [hunk 2]
```diff
@@ -170,102 +177,85 @@ void ExecutionManager::startApplication(const ProcessInfo& process)
...
             << " received"
             << std::endl;
-
-  return kj::READY_NOW;
 }
 
-::kj::Promise<void>
-ExecutionManager::register_(RegisterContext context)
+bool
+ExecutionManager::registerMachineStateClient(pid_t processId, string appName)
```
Comment (adklimkiew) [inline applications/ExecutionManager/src/execution_manager.cpp]: This method is not needed in EM. The whole register logic should be moved outside EM (to server).

File: applications/ExecutionManager/src/execution_manager.cpp [hunk 3]
```diff
@@ -170,102 +177,85 @@ void ExecutionManager::startApplication(const ProcessInfo& process)
...
+            << " registration failed"
+            << std::endl;
+
+  return false;
 }
 
-::kj::Promise<void>
-ExecutionManager::getMachineState(GetMachineStateContext context)
+MachineState
+ExecutionManager::getMachineState(pid_t processId) const
```
Comment (adklimkiew) [inline applications/ExecutionManager/src/execution_manager.cpp]: Why do you need arg processid?

File: applications/ExecutionManager/src/execution_manager_server.cpp [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+::kj::Promise<void>
+ExecutionManagerServer::register_(RegisterContext context)
+{
+  string newMachineClient = context.getParams().getAppName();
+  pid_t applicationPid = context.getParams().getPid();
+
+  if (m_em.registerMachineStateClient(applicationPid, newMachineClient))
+  {
+    context.getResults().setResult(StateError::K_SUCCESS);
```
Comment (adklimkiew) [inline applications/ExecutionManager/src/execution_manager_server.cpp]: You should call it once and use local variable for result with default value

File: applications/ExecutionManager/src/execution_manager_server.cpp [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+{
+  string state = context.getParams().getState().cStr();
+  pid_t applicationPid = context.getParams().getPid();
+
+  if (m_em.setMachineState(applicationPid, state))
+  {
+    context.getResults().setResult(StateError::K_SUCCESS);
+  }
+
+  context.getResults().setResult(StateError::K_INVALID_STATE);
```
Comment (adklimkiew) [inline applications/ExecutionManager/src/execution_manager_server.cpp]: You setResult twice. Incorrectly

File: applications/ExecutionManager/src/execution_manager.cpp [hunk 6]
```diff
@@ -170,102 +177,85 @@ void ExecutionManager::startApplication(const ProcessInfo& process)
...
   if (m_machineStateClientPid == -1 ||
-      m_machineStateClientPid == newMachineClientAppPid)
+      m_machineStateClientPid == processId)
   {
-    m_machineStateClientPid = newMachineClientAppPid;
-    m_machineStateClientAppName = newMachineClientAppName;
-
-    context.getResults().setResult(StateError::K_SUCCESS);
+    m_machineStateClientPid = processId;
+    m_machineStateClientAppName = appName;
```
Comment (adklimkiew) [inline applications/ExecutionManager/src/execution_manager.cpp]: Why do you need these paramaters?

**Pull request merged**